src/main/scala/lms/thirdparty/random_data.scala

```diff
@@ -0,0 +1,44 @@
+package lms.thirdparty
+
+import lms.core._
+import lms.util._
+import lms.core.stub._
+import lms.core.Backend._
+import lms.core.virtualize
+import lms.core.utils.time
+import lms.macros.SourceContext
+
+object RandomDataTypeLess {
+  import BaseTypeLess._
+  import PrimitiveTypeLess._
+  import CLibTypeLess._
+
+  // FIXME(feiw) add support for this:
+  /* Intializes random number generator */
+  // srand((unsigned) time(&t));
+
+  def rand_int(implicit __pos: SourceContext): INT = INT(CMACRO("rand()", manifest[Int]))
```

Comment:
Add a `rand_int` that accepts an upper bound:
```suggestion
  def rand_int(implicit __pos: SourceContext): INT = INT(CMACRO("rand()", manifest[Int]))
  def rand_int(ub: Int)(implicit __pos: SourceContext): INT = INT(CMACRO(s"(rand() % $ub) + 1", manifest[Int]))
```

Reply:
Added